poc/flp_pine.py

```diff
@@ -549,3 +635,35 @@ def chunk_count(chunk_length, length):
 
 def encode_float(x: float, num_frac_bits: int) -> int:
     return math.floor(x * 2**num_frac_bits)
+
+
+def construct_circuits(
+    field: type[F],
+    l2_norm_bound: int,
+    num_frac_bits: int,
+    dimension: int,
+    chunk_length: int,
+    chunk_length_norm_equality: int,
+    alpha: float = ALPHA,
+    num_wr_checks: int = NUM_WR_CHECKS,
+    num_wr_successes: int = NUM_WR_SUCCESSES
+) -> tuple[PineValidSquaredNormEqualityCheck[F], PineValid[F]]:
+    return PineValidSquaredNormEqualityCheck(
+        field,
+        l2_norm_bound,
+        num_frac_bits,
+        dimension,
+        chunk_length_norm_equality,
+        alpha,
+        num_wr_checks,
+        num_wr_successes,
+    ), PineValid(
+        field,
+        l2_norm_bound,
+        num_frac_bits,
+        dimension,
+        chunk_length,
+        alpha,
+        num_wr_checks,
+        num_wr_successes,
+    )
```

Comment:
Idea: We could move the code from `PineValiBase.__init__` here. Then all each circuit's constructor needs to do is copy the parameters it needs and construct the ones that are specific to it.

Reply:
Both critics share lots of parameters. Unless we want these two circuits to have different `MEAS_LEN`,  `PineValidNormEquality` has to carry `num_wr_checks`. So goes to other parameters.

What parameters do you want to split out of base class?


The base class should initializer the shared parameters. Having some external method to do this work is not a good practice.

And each constructor of the subclass already does "construct the ones that are specific to it." as of now.